tests/slate/test_assemble_tensors.py

```diff
@@ -100,17 +100,12 @@ def test_assemble_matrix_into_tensor(mesh):
 
 
 def test_mixed_coefficient_tensor(mesh):
-    V = FunctionSpace(mesh, "CG", 1)
-    U = FunctionSpace(mesh, "DG", 0)
-    W = V * U
+    V = FunctionSpace(mesh, "DG", 0)
+    W = V * V
     f = Function(W)
+    g, h = f.split()
     f.assign(1)
-    u = TrialFunction(V)
-    v = TestFunction(V)
-    T = Tensor(f[0] * u * v * dx)
-    ref = assemble(f[0] * u * v * dx)
-
-    assert np.allclose(assemble(T).M.values, ref.M.values)
+    assert np.allclose(assemble(Tensor((g + f[0] + h + f[1])*dx)), 4.0)
```

Comment:
Oh, I hadn't spotted that there was already a test.  I guess you could have kept the existing one and added this?